File: namoo.tutorial.JavaScript/js/Student.js
function Student(ssn,name,korean,english,math){ // Function 프로토타입 객체가 부모
  //this ={}; 눈에는 안보이지만 자동적으로 만들어져 있음
  //속성 
  this.ssn=ssn;
  this.name=name;
  this.korean=korean;
  this.english=english;
  this.math=math;
  //return this 도 자동으로 해줌
}
//메소드를 프로토타입에 추가
Student.prototype.getsum = function(){
  return this.korean + this.english + this.math;
}

Student.prototype.getAverage = function(){
  return this.getsum()/3 ;
}

Student.prototype.toString = function(){
  return `${this.ssn},${this.name},${this.korean},${this.english},${this.math},${this.getsum()},${this.getAverage()}`;
} //모든 속성값 나오게 만듬

Student.prototype.schoolName = "Korea 중학교";

export {Student};

File: maven.web/src/main/java/namoo/web/sts/dto/Student.java
package namoo.web.sts.dto;

public class Student {
	private int ssn;
	private String name;
	private int korean;
	private int math;
	private int english;
	
	

	public Student() {
	}

	public Student(int ssn, String name, int korean, int math, int english) {
		this.ssn = ssn;
		this.name = name;
		this.korean = korean;
		this.math = math;
		this.english = english;
	}
	
	public int getSsn() {
		return ssn;
	}
	public void setSsn(int ssn) {
		this.ssn = ssn;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public int getKorean() {
		return korean;
	}
	public void setKorean(int korean) {
		this.korean = korean;
	}
	public int getMath() {
		return math;
	}
	public void setMath(int math) {
		this.math = math;
	}
	public int getEnglish() {
		return english;
	}
	public void setEnglish(int english) {
		this.english = english;
	}
	@Override
	public String toString() {
		return "Student [ssn=" + ssn + ", name=" + name + ", korean=" + korean + ", math=" + math + ", english="
				+ english + "]";
	}
	public int avg() {
		return (this.english+ this.math+this.korean)/3; 
	}
	
	
	
}
